### ISSUE TITLE
Add IMS preprocessing code to GDASApp

### ISSUE THREAD

**Issue by CoryMartin-NOAA** [NOAA-EMC/GDASApp#198]
Add and test the IMS preprocessing code to GDASApp.

This includes:

- [x] Having GDASApp clone the repository like the existing JEDI repositories
- [x] Build the code as part of the complete build process
- [x] Test the code offline to ensure it is running as intended with the new build system
- [x] Add a ctest to verify the IMS preprocessing code does not become broken

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 is this work being done in the branch [feature/land-imsproc](https://github.com/NOAA-EMC/GDASApp/tree/feature/land-imsproc)?

**Comment by CoryMartin-NOAA** [comment]
This issue *may* depend on #201. @jiaruidong2017 depending on the size of the input test data we either will do 1) stage the test files in the test data tar ball or 2) keep them in a centralized place on Hera and Orion for testing.

**Comment by jiaruidong2017** [comment]
> @jiaruidong2017 is this work being done in the branch [feature/land-imsproc](https://github.com/NOAA-EMC/GDASApp/tree/feature/land-imsproc)?

@CoryMartin-NOAA I am in the land team meeting from 2 to 3pm. Yes, the IMS_proc work will be done in the branch feature/land-imsproc.

**Comment by CoryMartin-NOAA** [comment]
Note due to a recent PR there may be a build issue, where the IMS calcfIMS exe is not built, if so, modify this line:
https://github.com/NOAA-EMC/GDASApp/blob/ce1a55355278e951b292b6ba61ba0ba64775861d/build.sh#L114
to include `land-imsproc` in it

**Comment by jiaruidong2017** [comment]
> Note due to a recent PR there may be a build issue, where the IMS calcfIMS exe is not built, if so, modify this line:
> 
> https://github.com/NOAA-EMC/GDASApp/blob/ce1a55355278e951b292b6ba61ba0ba64775861d/build.sh#L114
> 
> 
> to include `land-imsproc` in it

Thanks @CoryMartin-NOAA, and I will rebuild it.

**Comment by jiaruidong2017** [comment]
> This issue _may_ depend on #201. @jiaruidong2017 depending on the size of the input test data we either will do 1) stage the test files in the test data tar ball or 2) keep them in a centralized place on Hera and Orion for testing.

@CoryMartin-NOAA What is the maximum limit if the test data tar ball is used? Do we have some tools to change data into low resolution or subset the observation data file?

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 there's no 'rule' on file size for the tar ball, but the current size of the dataset is ~30MB, and I would be hesitant to do more than double that for any one test, so my guess is C48 will even be way too big (but I'm not sure).

The input file, it depends on what the format is, there are several netCDF tools for subsetting, and I think ones exist for BUFR also.

**Comment by jiaruidong2017** [comment]
I tested the IMS processing code offline under the GDASApp environment, and it successfully generated the results at C48 resolution. 

**IMS OBS file:** (file size=567,046)
/scratch2/NCEPDEV/land/data/DA/snow_ice_cover/IMS/2021/ims2021083_4km_v1.3.nc

**IMS Index file:** (File size=4.1 MB)
/scratch2/NCEPDEV/land/data/DA/snow_ice_cover/IMS/index_files/IMS4km_to_FV3_mapping.oro_C48.mx100.nc 

For the next step, add a ctest in the test/land/CMakeLists.txt:

```
add_test(NAME test_gdasapp_land-imsproc
         COMMAND ${PROJECT_SOURCE_DIR}/test/land/submit_job_imsproc.sh ${PROJECT_BINARY_DIR} ${PROJECT_SOURCE_DIR}
         WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/land/)
```
A test script `submit_job_imsproc.sh` is put at test/land/ directory to prepare and initiate this ctest.

**Comment by CoryMartin-NOAA** [comment]
Closed through #216
